ui_tests/splunk_search.py

```diff
@@ -0,0 +1,266 @@
+    """
+    Send a request to a Splunk instance to create an index
+    @param: index (index to be deleted)
+    @param: url (splunkd rest api url)
+    @param: user (splunk username)
+    @param: password (splunk password)
+    returns True/False
+    """
+
+    search_url = "{}/services/data/indexes/{}?output_mode=json".format(url, index)
+    logger.debug("requesting: %s", search_url)
+    data = {"name": index}
+
+    create_job = _requests_retry_session().post(
+        search_url, auth=(user, password), verify=False, data=data
+    )
+
+    if create_job.status_code == 201:
+        logger.info("The index: %s successfully created", index)
+    elif create_job.status_code == 409:
+        logger.info("The index: %s already exits", index)
+    else:
+        logger.info(
+            "The index: {} not created, exit code is {}".format(
+                index, create_job.status_code
+            )
+        )
+        return False
+
+    return True
+
+
+def _collect_events(query, start_time, end_time, url="", user="", password=""):
+    """
+    Collect events by running the given search query
+    @param: query (search query)
+    @param: start_time (search start time)
+    @param: end_time (search end time)
+    returns events
+    """
+
+    search_url = "{}/services/search/jobs?output_mode=json".format(url)
+    logger.debug("requesting: %s", search_url)
+    data = {
+        "search": query,
+        "earliest_time": start_time,
+        "latest_time": end_time,
+    }
+    logger.debug("SEARCH DATA: {}".format(data))
+    create_job = _requests_retry_session().post(
+        search_url, auth=(user, password), verify=False, data=data
+    )
+    _check_request_status(create_job)
+
+    json_res = create_job.json()
+    job_id = json_res["sid"]
+    events = _wait_for_job_and__get_events(job_id, url, user, password)
+
+    return events
+
+
+def _collect_metrics(
+    start_time, end_time, url="", user="", password="", index="", metric_name=""
+):
+    """
+    Verify metrics by running the given api query
+    @param: dimension (metric dimension)
+    @param: metric_name (metric name)
+    @param: start_time (search start time)
+    @param: end_time (search end time)
+    returns events
+    """
+    api_url = (
+        url
+        + "/services/catalog/metricstore/dimensions/host/values?filter=index%3d"
+        + index
+        + "&metric_name="
+        + metric_name
+        + "&earliest="
+        + start_time
+        + "&latest="
+        + end_time
+        + "&output_mode=json"
+    )
+    logger.debug("requesting: %s", api_url)
+
+    create_job = _requests_retry_session().get(
+        api_url, auth=(user, password), verify=False
+    )
+
+    _check_request_status(create_job)
+
+    json_res = create_job.json()
+
+    events = json_res["entry"]
+    # logger.info('events: %s', events)
+
+    return events
+
+
+def _wait_for_job_and__get_events(job_id, url="", user="", password=""):
```

Comment:
there is double _ in the name before `get`, please change it to one _

Reply:
done